internal/pkg/cli/env_package.go

```diff
@@ -152,23 +156,25 @@ func (o *packageEnvOpts) Execute() error {
 	if err != nil {
 		return fmt.Errorf("get caller principal identity: %v", err)
 	}
-	deployer, err := o.newEnvPackager()
+	packager, err := o.newEnvPackager()
 	if err != nil {
 		return err
 	}
-	if err := deployer.Validate(mft); err != nil {
+	if err := packager.Validate(mft); err != nil {
 		return err
 	}
-	var urls map[string]string
+	var uploadArtifactsOut deploy.UploadEnvArtifactsOutput
 	if o.uploadAssets {
-		urls, err = deployer.UploadArtifacts()
+		out, err := packager.UploadArtifacts()
```

Comment:
Let me know if this train of thought makes sense to you too:
1. Since all the hashes and the URL of these artifacts are deterministic, I wonder if this signature makes more sense:
   ```
   err := packager.UploadArtifacts()
   ```
2. Instead `GenerateCloudFormationTemplate` holds the responsibility of `parseAddons` and calling `artifactpath.EnvironmentAddons` at all times so that the template output is always the same regardless of using the `--upload-assets` flag

Reply:
It's okay if this is not done part of this PR as it's larger change, let me know what you think!

The extensions project would need to make the change above anyway, so I feel like it'd be nice to introduce it